src/server/req_quejob.c

```diff
@@ -271,27 +271,40 @@ get_server_index(void)
  * 			Note: Consider directly modifying get_next_svr_sequence_id() if reservations
  * 			also get sharded for multi-server
  *
- * @param[out]	jidbuf - buffer to fill job id in
+ * @param[out]	idbuf - buffer to fill job/resv id in
  * @param[in]	clusterid - cluster name (PBS_SERVER)
- * @param[in]	jtype - job type (0 for normal job, 1 for job array)
+ * @param[in]	objtype - object type (0 for normal job, 1 for job array, 2 for reservation)
+ * @param[in]	resv_char - character representing type of reservation
  *
  * @return	int
  * @retval	0 for Success
  * @retval	1 for Failure
  */
 static int
-generate_jobid(char *jidbuf, char *clusterid, int jtype)
+generate_objid(char *idbuf, char *clusterid, int objtype, char resv_char)
 {
 	static int svr_id = -1;
 
-	if (jidbuf == NULL || server_name == NULL)
+	if (idbuf == NULL || server_name == NULL)
 		return 1;
 
 	if (get_num_servers() <= 1) { /* single server setup */
-		if (jtype == 0)
-			sprintf(jidbuf, "%lld.%s", next_svr_sequence_id, clusterid);
-		else
-			sprintf(jidbuf, "%lld[].%s", next_svr_sequence_id, clusterid);
+		if (clusterid != NULL) {
```

Comment:
clusterid will never be NULL, we don't need to handle that case (here and below)